system/xmllib/xpathprocessor.hpp

```diff
@@ -81,4 +81,22 @@ class CXpathContextScope : CInterface
 extern "C" XMLLIB_API ICompiledXpath* compileXpath(const char * xpath);
 extern "C" XMLLIB_API IXpathContext*  getXpathContext(const char * xmldoc, bool strictParameterDeclaration);
 
+interface IEsdlScriptContext : extends IInterface
+{
+    virtual IXpathContext* createXpathContext(const char *section, bool strictParameterDeclaration) = 0;
+    virtual void *queryEspContext() = 0;
+    virtual void setContent(const char *section, const char *xml) = 0;
+    virtual void setContent(const char *section, IPropertyTree *tree) = 0;
+    virtual void setAttribute(const char *section, const char *name, const char *value) = 0;
+    virtual const char *queryAttribute(const char *section, const char *name) = 0;
+    virtual const char *getAttribute(const char *section, const char *name, StringBuffer &s) = 0;
+    virtual const char *getXPathString(const char *xpath, StringBuffer &s) const = 0;
+    virtual __int64 getXPathInt64(const char *xpath, __int64 dft=0) const = 0;
+    virtual bool getXPathBool(const char *xpath, bool dft=false) const = 0;
```

Comment:
I know this naming fits existing code style (like PTree), but I'm not a fan. My preference would be to follow the design of IEspSecureContext. It would handle unsigned versus signed, or floating point versus integer, or short versus long.

A similar change could be made to getAttribute and setAttribute.

Reply:
Can consider refactoring in the future.  At least this is following a precedent.  I don't think this is currently an interface that we can't easily change.